fast-list.js

```diff
@@ -173,28 +173,44 @@ FastList.prototype = {
       geo.forward = false;
     }
 
-    if (geo.topPosition === 0 && this._previousTop !== 0) {
+    if (geo.topPosition === 0 && previousTop !== 0) {
       this.els.container.dispatchEvent(new CustomEvent('top-reached'));
     }
 
+    var onTop = geo.topPosition === 0;
+    var atBottom = geo.topPosition ===
+      (this.source.getFullHeight() - geo.viewportHeight);
+
+    // Full stop, forcefuly idle
+    if (onTop || atBottom) {
+      geo.busy = false;
+      geo.idle = true;
+      return;
+    }
+
     var moved = Math.abs(distanceSinceLast);
-    var fastScrolling = this._previousFast;
 
-    if (!fastScrolling  && moved > geo.viewportHeight * 2) {
-      fastScrolling = true;
+    var previousBusy = geo.busy;
+    var previousIdle = geo.idle;
+
+    if (!previousBusy  && moved > geo.viewportHeight * 2) {
+      previousBusy = true;
     }
 
-    if (fastScrolling && moved > 0 && moved < geo.viewportHeight * 0.5) {
-      fastScrolling = false;
+    if (previousBusy && moved > 0 && moved < geo.viewportHeight * 0.5) {
+      previousBusy = false;
```

Comment:
NIT: This naming is a little confusing, as what we're really saying is `currentlyBusy = false`. Could we set `geo.busy = false` here instead and not change the value of `previouslyBusy`? Same applies for `previouslyIdle`.